src/figuresClasses.ts

```diff
@@ -1,11 +1,76 @@
-export interface Figure {}
+'use strict';
 
-export class Triangle implements Figure {}
+type Shapes = 'triangle' | 'circle' | 'rectangle';
 
-export class Circle implements Figure {}
+type Colors = 'red' | 'green' | 'blue';
 
-export class Rectangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
 
-export function getInfo(figure): string {
-  return typeof figure;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Colors,
+    public a: number,
```

Comment:
Let's avoid naming variables as 'a | b | c | p' - it's not an alphabet, each variable actually means something. 

If shape can be named as 'shape', why can't we use 'sideA' instead of just 'a'? :)

Reply:
Good question 😅
I renamed them